frontend/src/components/SubmissionWidget.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { SubmissionWidgetProps } from "../models/submission";
+
+const SubmissionWidget: React.FC<SubmissionWidgetProps> = ({
+  question,
+  globalDocs,
+}) => {
+  const [activeTab, setActiveTab] = useState("submission");
+  const [docsTab, setDocsTab] = useState("question");
+
+  const handleTabSwitch = (tab: "submission" | "docs") => {
+    setActiveTab(tab);
+  };
+
+  const handleDocsTabSwitch = (tab: "question" | "global") => {
+    setDocsTab(tab);
+  };
+
+  /* 
+  Generate the link for accessing the backend route.
+  This function will return a route depending on whether
+  the doc in question is a global doc or a question doc.
+  */
+  const generateDocRoute = (docTitle: string, isGlobal: boolean) => {
+    return isGlobal
+      ? `/docs/global/${docTitle}`
+      : `/docs/question/${question.num}/${docTitle}`;
+  };
+
+  return (
+    <section className="w-full max-w-lg h-[95vh] mx-auto bg-white rounded-lg shadow-md p-4 mt-1 mb-2 lg:ml-auto lg:mr-4 relative">
+      <div className="flex border-b">
+        <button
+          onClick={() => handleTabSwitch("submission")}
+          className={`w-1/2 py-2 text-center ${
```

Comment:
This is a super cool way to handle different styles on a condition!

Reply:
Thanks dude!